plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/segmented/RocksDBColumnarKeyValueSnapshot.java

```diff
@@ -167,14 +199,12 @@ public void clear(final SegmentIdentifier segment) {
   public boolean containsKey(final SegmentIdentifier segment, final byte[] key)
       throws StorageException {
     throwIfClosed();
-    return snapTx.get(segment, key).isPresent();
+    return get(segment, key).isPresent();
   }
 
   @Override
   public void close() throws IOException {
-    if (closed.compareAndSet(false, true)) {
-      snapTx.close();
-    }
+    closed.set(true);
```

Comment:
I think we need to modify the close in order to close the snapshot and the readoptions
```java
public void close() throws IOException {
    if (closed.compareAndSet(false, true)) {
      closed.set(true);
      readOptions.close();
      snapshot.close();
    }
  }
  ```